src/adaptermanager.js

```diff
@@ -19,53 +19,61 @@ let _s2sConfig = config.getConfig('s2sConfig');
 
 var _analyticsRegistry = {};
 
-function getBids({bidderCode, auctionId, bidderRequestId, adUnits}) {
-  return adUnits.map(adUnit => {
-    return adUnit.bids.filter(bid => bid.bidder === bidderCode)
-      .map(bid => {
-        let sizes = adUnit.sizes;
-        if (adUnit.sizeMapping) {
-          let sizeMapping = mapSizes(adUnit);
-          if (sizeMapping === '') {
-            return '';
-          }
-          sizes = sizeMapping;
-        }
+function getBids({bidderCode, auctionId, bidderRequestId, adUnits, labels}) {
+  function getLabels(obj, requestLabels) {
+    if (obj.labelAll) {
+      return {labelAll: true, labels: obj.labelAll, requestLabels};
```

Comment:
in my experience, creating objects on the fly like this is better done more formally, at least in a function were you can use @param and also reduces chances to code to a single place.

Reply:
Updated some names and added a typedef to help.  I tried to use a constructor function but my IDEs linter was not having it, claiming a type mismatch; so left as factory function.